Floofbot/Modules/Fun.cs

```diff
@@ -168,5 +168,172 @@ public async Task RequestCatFact()
             }
             await Context.Channel.SendMessageAsync(fact);
         }
+
+        [Command("minesweeper")]
+        [Summary("Minesweeper minigame")]
+        public async Task Minesweeper([Summary("grid x")]int gridx, [Summary("grid y")]int gridy, [Summary("bomb count")]int bombs)
+        {
+           
+            //no negative numbers!
+            if(gridx < 1 || gridy < 1 || bombs < 0)
+            {
+                await Context.Channel.SendMessageAsync("Invalid grid size or bomb count");
+                return;
+            }
+            
+            //limits the size of the board
+            if(gridx > 10 || gridy > 10)
+            {
+                await Context.Channel.SendMessageAsync("Max Grid Size: 10 x 10");
+                return;
+            }
+            else if(bombs >= gridy * gridx)
+            {
+                await Context.Channel.SendMessageAsync("Too many bombs!");
+                return;
+            }
+
+            Gameboard game = new Gameboard(gridx, gridy, bombs);
+            EmbedBuilder builder = new EmbedBuilder();
+            builder.Title = ":bomb: Minesweeper";
+            builder.Color = EMBED_COLOR;
+            builder.Description = game.getBoard();
+
+            await Context.Channel.SendMessageAsync("", false, builder.Build());
+        }
+
+        public class Gameboard
+        {
+            private string[,] grid;
+            private int x;
+            private int y;
+            private int bombCount;
+
+            public Gameboard(int _x, int _y, int _bombCount)
+            {
+                x = _x;
+                y = _y;
+                bombCount = _bombCount;
+                grid = new string[x, y];
+                plantBombs();
+
+
+                for (int i = 0; i < x; i++)
+                {
+                    for (int j = 0; j < y; j++)
+                    {
+                        if (grid[i, j] != "||:bomb:||")
+                        {
+                            grid[i, j] = getBombCount(i, j).ToString();
+                        }
+                    }
+                }
+            }
+
+            public int getBombCount(int _x, int _y)
+            {
+                int count = 0;
+
+                for (int i = -1; i <= 1; i++)
+                {
+                    for (int j = -1; j <= 1; j++)
+                    {
+                        if (!(i == 0 && j == 0))
+                        {
+                            try
+                            {
+                                if (grid[_x + i, _y + j] == "||:bomb:||")
+                                {
+                                    count++;
+                                }
+                            }
+                            catch { } //stupid but works I  guess
+                        }
```

Comment:
does this need to be in a try catch block? you can also use if, else continue;

Reply:
Yes, because this is doing a naive out of bounds check on the grid. I think it's worth commenting that rather than just calling it something stupid, though.

Reply:
In that case I think I would prefer making the boundaries a variable that can be referenced than using errors